CoffeeShop/src/main/java/item/ItemList.java

```diff
@@ -0,0 +1,113 @@
+package item;
+
+import interfaces.EntityList;
+import java.util.HashMap;
+import utils.ItemCategory;
+import exceptions.DuplicateItemIDException;
+
+/**
+ * @author Fraser Holman
+ *
+ * Class represents a list of all avaible items that can be ordered
+ *
+ * Contains a hashmap of all the items
+ */
+
+public class ItemList implements EntityList<Item, String> {
+    private HashMap<String, Item> items;
+
+    /**
+     * Initialises the hashmap to contain the items
+     */
+    public ItemList() {
+        items = new HashMap<String, Item>();
+    };
+
+    /**
+     * Adds an item to the hashmap
+     *
+     * @param item The item to be added to the hashmap
+     */
+    @Override
+    public void add(Item item) throws DuplicateItemIDException {
+        if (items.containsKey(item.getItemID())) {
+            throw new DuplicateItemIDException("Item ID is not unique");
+        }
+        else items.put(item.getItemID(), item);
+    }
+
+    /**
+     *
+     * @param ID
+     */
+    @Override
+    public void remove(String ID) {
+        if (items.containsKey(ID)) {
+            items.remove(ID);
+            return;
+        }
+        throw new IllegalArgumentException(ID + " is not a valid item ID");
+    }
```

Comment:
```suggestion
    @Override
    public boolean remove(String ID) {
        return items.remove(ID) != null;
    }
```
here is a simplified approach, links in with the order list one too but basically .remove should return true or false i think, depending if the operation fails or not. Then we can let the exception 'bubble up' and handle it wherever we call this